### PR TITLE
Mgmt update

### LINKED ISSUE #4: Update the rest interface to match the Whisk CLIs expectation
We need to have a rest interface that matches the rest interface provided and used by the OpenWhisk CLI. This includes the tenants piece. In this scenario, we could ideally take the namespace and instance, and return it concatenated to be used later when creating resources. We will also need to modify our rest interface for resources to match.

### PR DESCRIPTION
**Pull request by alexsong93**
#Modify the management interface to match that of `gateway-director-management-interface`, including `/APIs` and `/Tenants`. @mhamann @lostinthestory PTAL. Closes #4

### CODE REVIEW

File: api-gateway-config/conf.d/management_apis.conf [hunk 1]
```diff
@@ -33,19 +33,37 @@ server {
     access_log /var/log/api-gateway/access.log platform;
     error_log /var/log/api-gateway/mgmt_error.log debug;
 
-    location /resources {
+    location /v1/APIs {
```
Comment (mhamann) [inline api-gateway-config/conf.d/management_apis.conf]: should be all lowercase

File: api-gateway-config/conf.d/management_apis.conf [hunk 2]
```diff
@@ -33,19 +33,37 @@ server {
...
+                mgmt.deleteAPI()
+            else
                 ngx.status = 400
-                ngx.say("Use PUT")
+                ngx.say("Invalid verb")
+            end
+        }
+    }
+
+    location /v1/Tenants {
```
Comment (mhamann) [inline api-gateway-config/conf.d/management_apis.conf]: lowercase

File: api-gateway-config/scripts/lua/lib/redis.lua [hunk 3]
```diff
@@ -78,44 +82,83 @@ function _M.close(red)
...
+---------------------------
+
+--- Add API to redis
+-- @param red Redis client instance
+-- @param id id of API
+-- @param apiObj the api to add
+function _M.addAPI(red, id, apiObj)
+  local ok, err = red:hset("apis", id, apiObj)
+  if not ok then
+    request.err(500, utils.concatStrings({"Failed adding API to redis: ", err}))
```
Comment (mhamann) [inline api-gateway-config/scripts/lua/lib/redis.lua]: Probably best to return a generic error here like `Failed to save the API`, while still logging the actual reason. Same comment for the other pieces of this API: `Failed to retrieve APIs`, etc.

File: api-gateway-config/scripts/lua/lib/redis.lua [hunk 4]
```diff
@@ -169,6 +212,59 @@ function _M.deleteResource(red, key, field)
...
+-----------------------------
+
+--- Add tenant to redis
+-- @param red Redis client instance
+-- @param id id of tenant
+-- @param tenantObj the tenant to add
+function _M.addTenant(red, id, tenantObj)
+  local ok, err = red:hset("tenants", id, tenantObj)
+  if not ok then
+    request.err(500, utils.concatStrings({"Failed adding tenant to redis: ", err}))
```
Comment (mhamann) [inline api-gateway-config/scripts/lua/lib/redis.lua]: Same comment here as for the APIs--make things generic, but also log the message to someone can see in the logs what went wrong.

File: api-gateway-config/scripts/lua/lib/utils.lua [hunk 5]
```diff
@@ -75,9 +75,20 @@ function convertTemplatedPathParam(m)
   return concatStrings({"(?<path_" , x , ">([a-zA-Z0-9\\-\\s\\_\\%]*))"})
 end
 
+--- Generate random uuid
+function uuid()
+  local template ='xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'
+  math.randomseed(os.time())
```
Comment (mhamann) [inline api-gateway-config/scripts/lua/lib/utils.lua]: Good job seeding the randomizer so we don't end up with duplicates.

File: api-gateway-config/scripts/lua/management.lua [hunk 6]
```diff
@@ -28,127 +28,460 @@ local filemgmt = require "lib/filemgmt"
...
+  end
+  -- Additional check f or tenantId
+  if field == "tenantId" then
+    local tenant = redis.getTenant(red, object)
+    if tenant == nil then
+      return false, { statusCode = 404, message = utils.concatStrings({"Unknown tenant id ", object }) }
+    end
+  end
+  -- Additional checks for resource object
+  if field == "resources" then
```
Comment (mhamann) [inline api-gateway-config/scripts/lua/management.lua]: Seems like a lot of nesting here. We can deliver as-is, but I think we should revisit in the future and figure out a better, more elegant way of doing this. Potentially leads to side effects.

File: api-gateway-config/scripts/lua/management.lua [hunk 7]
```diff
@@ -28,127 +28,460 @@ local filemgmt = require "lib/filemgmt"
...
+      operations = v
+    end
+  end
+  local resourceObj = redis.generateResourceObj(operations, apiId)
+  redis.createResource(red, redisKey, REDIS_FIELD, resourceObj)
+  filemgmt.createResourceConf(BASE_CONF_DIR, tenantId, gatewayPath, resourceObj)
+end
+
+--- Get one or all APIs from the gateway
+-- PUT http://0.0.0.0:9000/APIs
```
Comment (mhamann) [inline api-gateway-config/scripts/lua/management.lua]: PUT?

File: api-gateway-config/scripts/lua/management.lua [hunk 8]
```diff
@@ -28,127 +28,460 @@ local filemgmt = require "lib/filemgmt"
...
+  }
+  tenantObj = cjson.encode(tenantObj)
+  redis.addTenant(red, uuid, tenantObj)
+  redis.close(red)
+  ngx.header.content_type = "application/json; charset=utf-8"
+  request.success(200, tenantObj)
+end
+
+--- Get one or all tenants from the gateway
+-- PUT http://0.0.0.0:9000/Tenants
```
Comment (mhamann) [inline api-gateway-config/scripts/lua/management.lua]: PUT?

**Comment by alexsong93** [comment]
@mhamann I've made the necessary changes in your comments

**Pull request merged**